modules/@apostrophecms/notification/ui/apos/apps/ApostropheNotification.js

```diff
@@ -0,0 +1,124 @@
+import Vue from 'apostrophe/vue';
+
+export default function() {
+  /* eslint-disable no-new */
+  return new Vue({
+    el: '#apos-notification',
+    data () {
+      return {
+        notifications: []
+      };
+    },
+    async mounted() {
+      apos.notify = async function(message, options) {
+        const strings = [];
+        let i = 1;
+        let index = 0;
+        while (true) {
+          index = message.indexOf('%s', index);
+          if (index === -1) {
+            break;
+          }
+          // Don't match the same one over and over
+          index += 2;
+          if ((i >= arguments.length) || ((typeof (arguments[i]) === 'object'))) {
+            throw new Error('Bad notification call: number of %s placeholders does not match number of string arguments after message');
+          }
+          strings.push(arguments[i++]);
+        }
+        if ((i === (arguments.length - 1)) && (typeof (arguments[i]) === 'object')) {
+          options = arguments[i++];
+        } else {
+          options = {};
+        }
+
+        if (i !== arguments.length) {
+          throw new Error('Bad notification call: number of %s placeholders does not match number of string arguments after message');
+        }
+
+        if (options.dismiss === true) {
+          options.dismiss = 5;
+        }
+
+        // Send it to the server, which will send it back to us via
+        // the same long polling mechanism that allows it to reach
+        // other tabs, and allows server-sent notifications to
+        // reach us
+
+        await apos.http.post(apos.notification.action, {
+          body: {
+            message,
+            strings,
+            type: options.type,
+            dismiss: options.dismiss,
+            id: options.id
+          }
+        });
+      };
+
+      await this.poll();
+    },
+    methods: {
+      async dismiss(notificationId) {
+        await apos.http.patch(`${apos.notification.action}/${notificationId}`, {
+          body: {
+            dismissed: true
+          }
+        });
+        this.notifications = this.notifications.reduce((acc, cur) => {
+          if (cur._id !== notificationId) {
+            acc.push(cur);
+          }
+          return acc;
+        }, []);
+      },
+      async poll() {
+        try {
+          const latestTimestamp = this.notifications
+            .map(notification => notification.updatedAt)
+            .sort()
+            .reverse()[0];
+
+          const _id = this.notifications
+            .filter(notification => notification.updatedAt === latestTimestamp)
+            .map(notification => notification._id)
+            .join(',_id=');
```

Comment:
Don't send all the ids, send the ids with the newest value of `updatedAt` that you have seen. Including any you have dismissed that still match the newest value of `updatedAt`.

Reply:
Not sure to understand. In this code, I send the seen ids with the newest value of updatedAt. 

Why is it important to also send the dismissed ones with the same timestamp? As long as they are dismissed, they won't be displayed (or they will be hidden if they are multiple open tabs - case already taken into account few lines under). If I have to find the dismissed ids with a specific timestamp on the frontend side, I need to send actual dismissed notifications (currently, the `dismissed` array only contains `_id`s to save space).

Reply:
If you don't include dismissed notifications that happen to match the newest timestamp, and the user has dismissed the last notification, then the server will always find the latest dismissed notification and send it because it matches the timestamp. So we won't have long polling, just an immediate response and a fast infinite loop of queries to the server.

For this reason the server's response should indeed include the dismissed notifications, but notice that it won't be a lot of duplicate information because it will only be those whose timestamps are new enough. Maybe one, or two in the case of a very rare race condition (which unfortunately can happen, which is why we're doing all this (: )

Reply:
Ok, I understand the `latestTimestamp` should be found in `notifications` (and not in `notifications` + `dismissed`), but the `seenIds` should be found in `notifications` + `dismissed`. Correct?